runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -496,6 +507,13 @@ public void putAll(@Nonnull Map<? extends K, ? extends V> m) {
     @Mutator(name = "clear", reset = true)
     public void clear() {
         mainMap.clear();
+        try {
+            indexExecutor.submit(() -> null).get();
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)